test/libsolidity/SemanticTest.cpp

```diff
@@ -321,7 +355,7 @@ void SemanticTest::printSource(ostream& _stream, string const& _linePrefix, bool
 void SemanticTest::printUpdatedExpectations(ostream& _stream, string const&) const
 {
 	for (auto const& test: m_tests)
-		_stream << test.format("", true, false) << endl;
+		_stream << test.format("", !m_gasCostFailure, false);
```

Comment:
The docstring for the `format()` function says:
```c++
	/// Formats this function call test and applies the format that was detected during parsing.
	/// If _renderResult is false, the expected result of the call will be used, if it's true
	/// the actual result is used.
	/// If _highlight is false, it's formatted without colorized highlighting. If it's true, AnsiColorized is
	/// used to apply a colorized highlighting.
	/// If test expectations do not match, the contract ABI is consulted in order to get the
	/// right encoding for returned bytes, based on the parsed return types.
	/// Reports warnings and errors to the error reporter.
```

So if I understand this correctly, your change means to use the expected gas cost if gas check failed and actual gas cost if it passed. But if it passed, the actual cost must have been the same as the expectation. So it's always the expected cost and you could just set it to `false`.

This would actually make sense if there was a tolerance on the cost comparison but you're doing an exact comparison instead.

BTW, this is the only place outside of `runTest()` that uses `m_gasCostFailure` so if it turns out that you do change it to `false`, then I think the variable should be converted to a local variable.

Reply:
I wanted to call with false in case of gas cost failure, and true in case of regular expectations failure. This is used to print error messages and update expectations accordingly.. 
There was a wierd issue when function call expects failure, and if I pass `true` here in case of only gas failure it changes expectations in a way that translates strings to raw bytes. So, I wanted to make sure gas costs doesn't touch other expectations with this flag...